src/PowerShellEditorServices.Protocol/Server/DebugAdapter.cs

```diff
@@ -427,20 +428,27 @@ protected void Stop()
 
                     return;
                 }
+            }
+            else if (customPipeNameIsSet)
+            {
+                if (runspaceVersion.Version < _minVersionForCustomPipeName)
+                {
+                    await requestContext.SendErrorAsync(
+                        $"Attaching to a process with CustomPipeName is only available with PowerShell 6.2 and higher (current session is {runspaceVersion.Version}).");
+                    return;
+                }
 
-                // Clear any existing breakpoints before proceeding
-                await ClearSessionBreakpointsAsync();
-
-                // Execute the Debug-Runspace command but don't await it because it
-                // will block the debug adapter initialization process.  The
-                // InitializedEvent will be sent as soon as the RunspaceChanged
-                // event gets fired with the attached runspace.
-                int runspaceId = attachParams.RunspaceId > 0 ? attachParams.RunspaceId : 1;
-                _waitingForAttach = true;
-                Task nonAwaitedTask =
-                    _editorSession.PowerShellContext
-                        .ExecuteScriptStringAsync($"\nDebug-Runspace -Id {runspaceId}")
-                        .ContinueWith(OnExecutionCompletedAsync);
+                await _editorSession.PowerShellContext.ExecuteScriptStringAsync(
+                    $"Enter-PSHostProcess -CustomPipeName {attachParams.CustomPipeName}",
+                    errorMessages);
```

Comment:
Can you still get intellisense while in the attached process?  I would think that this would hold up the request queue, or does `Enter-PSHostProcess` return in the original runspace after attaching?

Reply:
There is so little time between `Enter-PSHostProcess` and the `Debug-Runspace` call bellow that it doesn't really matter. Debug-Runspace will for sure block any intellisense call until a breakpoint is hit. Then once a breakpoint is hit, it will work again but from the context of the breakpoint.

Reply:
Does that answer your question?

FWIW, I'm not changing that behavior in this PR. In the old version it still:

awaited `Enter-PSHostProcess`
not awaited `Debug-Runspace`

Reply:
Talked about this offline. We are good.